README.md

```diff
@@ -4,222 +4,1008 @@
+enum Mode
+{
+    normal,  // Even returns 1. Odd returns 0.
+    reversed // Even returns 0. Odd return 1.
+}
+
+@Command(null, "The default command.")
+struct DefaultCommand
+{
+    @CommandPositionalArg(0, "number", "The number to check.")
+    int number;
+
+    @CommandNamedArg("mode", "Which mode to use.")
+    Mode mode;
+
+    int onExecute()
+    {
+        if(this.mode == Mode.normal)
+            return number % 2 == 0 ? 1 : 0;
+        else
+            return number % 2;
+    }
+}
+```
+
+To start off, we create an `enum` called `Mode`, and give the members `normal` and `reversed`. JCLI knows how to map strings into enums.
+
+Next, inside of `DefaultCommand` we create a member field called `Mode mode;` that is decorated with the `CommandNamedArg` UDA.
+
+The first parameter is the name of the argument, which is actually important this time as this determines what name the user has to use.
+
+The second parameter is just the description.
+
+Then inside of `onExecute` we just check what `mode` was set to and do stuff based off of its value.
+
+Time to test! Remember to run `dub build` first.
+
+Let's have a quick look at the help text first, to see the changes being reflected:
+
+```bash
+$> ./mytool --help
+Usage: DEFAULT {0/number} [mode]
+
+Description:
+    The default command.
+
+Positional Args:
+    0,number                     - The number to check.
+
+Named Args:
+    --mode                       - Which mode to use.
+```
+
+And now let's test our functionality:
+
+```bash
+# JCLI supports most common argument styles.
+
+# Even (Normal)
+$> mytool 60 --mode normal
+Program exited with status code 1
+
+# Even (Reversed)
+$> mytool 60 --mode=reversed
+Program exited with status code 0
+
+# Bad value for mode
+$> mytool 60 --mode lol
+std.conv.ConvException@\src\phobos\std\conv.d(2817): Mode does not have a member named 'reverse'
+
+# Can safely assume Odd behaves properly.
+
+# Now, we haven't marked --mode as optional, so...
+$> mytool 60
+ERROR: The following required named arguments were not provided: ["mode"]
+Program exited with status code -1
+```
+
+We can see that `--mode` is working as expected, however notice that in the last case, the user isn't allowed to leave out `--mode` since it's not marked as optional.
+
+## Optional Arguments
+
+Optional arguments, as the name implies, are optional. Only Named arguments can be optional (technically, Positional arguments can be optional in certain use cases, but JCLI doesn't support that... yet).
+
+Inside of D's standard library - Phobos - is the module [std.typecons](https://dlang.org/phobos/std_typecons.html) which contains a type called [Nullable](https://dlang.org/phobos/std_typecons.html#Nullable).
+
+JCLI has special support for this type, as it is used to mark an argument as optional. This type is publicly imported by JCLI, so you don't have to import anything extra.
+
+Anyway, all we want to do is make our `mode` argument a `Nullable`, so JCLI knows it's optional:
```

Comment:
```suggestion
JCLI supports optional arguments through standard [Nullable](https://dlang.org/phobos/std_typecons.html#Nullable) type. Note that only Named arguments can be optional for now (technically, Positional arguments can be optional in certain use cases, but it's not supported ... yet).

So to make our `mode` argument optional, we need to make it `Nullable`:
```

Technically "optional" means that it can be omitted in command line and so inside a program it might be either "not set" (e.g. `null`) or to be set to some default value so consider this use case:
```
    @CommandNamedArg("mode", "Which mode to use.")
    @DefaultValue("normal")
    Mode mode;
```
or:
```d
enum {
    @CommandArgValue("normal", "Even returns 1. Odd returns 0.")
    @DefaultValue
    normal_mode,
    @CommandArgValue("reversed", "Even returns 0. Odd returns 1.")
    reverse_mode,
}
```

Reply:
Unless I'm missing something, [Nullable](https://dlang.org/phobos/std_typecons.html#Nullable) supports this does it not?

`Nullable.isNull` can be used to check if the user passed in a value or not.

`Nullable.get` as explained can be to get a default value if the user didn't pass a value in themselves.